android/src/main/java/com/reactcommunity/rndatetimepicker/RNDismissableDatePickerDialog.java

```diff
@@ -53,4 +60,79 @@ protected void onStop() {
       super.onStop();
     }
   }
+
+  private void fixSpinner(Context context, int year, int month, int dayOfMonth) {
+    if (Build.VERSION.SDK_INT == Build.VERSION_CODES.N) {
+      try {
+        // Get the theme's android:datePickerMode
+        final int MODE_SPINNER = 2;
+        Class<?> styleableClass = Class.forName("com.android.internal.R$styleable");
+        Field datePickerStyleableField = styleableClass.getField("DatePicker");
+        int[] datePickerStyleable = (int[]) datePickerStyleableField.get(null);
+
+        final TypedArray a =
+            context.obtainStyledAttributes(
+                null, datePickerStyleable, android.R.attr.datePickerStyle, 0);
+        Field datePickerModeStyleableField = styleableClass.getField("DatePicker_datePickerMode");
+        int datePickerModeStyleable = datePickerModeStyleableField.getInt(null);
+        final int mode = a.getInt(datePickerModeStyleable, MODE_SPINNER);
+        a.recycle();
+
+        if (mode == MODE_SPINNER) {
```

Comment:
thanks for the PR, looks like this condition can be moved up, to `if (Build.VERSION.SDK_INT == Build.VERSION_CODES.N)`, is that so? (same in the second class)

Reply:
sorry, I was being blind